xonsh/completer.py

```diff
@@ -78,16 +78,54 @@ def _normpath(p):
 
     return p
 
-class Completer(object):
-    """This provides a list of optional completions for the xonsh shell."""
+class KnownCommands(object):
 
     def __init__(self):
-        # initialize command cache
         self._path_checksum = None
         self._alias_checksum = None
         self._path_mtime = -1
         self._cmds_cache = frozenset()
+    
+    def all_commands(self):
+        """find all commands available in PATH
+        """
+        path = builtins.__xonsh_env__.get('PATH', [])
+        # did PATH change?
+        path_hash = hash(tuple(path))
+        cache_valid = path_hash == self._path_checksum
+        self._path_checksum = path_hash
+        # did aliases change?
+        al_hash = hash(tuple(sorted(builtins.aliases.keys())))
```

Comment:
Me culpa here, but Isn't possible for different instances of the frozenset with the same values to hash differently? Or am I being overly paranoid here?